### PR TITLE
Fix UML Diagram for README and Rust Docs. Setup Cargo Make task runner. Setup Cargo Watch.

### PR DESCRIPTION
**Pull request by ltfschoen**
I've fixed the UML diagram so that it generates correctly and is shown in the README file and in the Rust Docs. I've also integrated Cargo Make, which is awesome, as you just configure a default Makefile.toml for all your tasks, with much simpler usage. For example, to run all the implementations at once (i.e. proposer mode, notary mode, and both mode) just run `cargo make all`, or to run all the unit and integration tests for the main package and all the libraries just run `cargo make t`, or to run Cargo Watch so it detects whenever you change any code and automatically runs all unit and integration tests and shows if you've encountered any issues then just run it with `cargo make test` and leave that running in a terminal window whilst your writing all your code so you know you're not breaking anything.
And if you want to open the UML diagram just run `cargo make uml`.

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+]
+
+[tasks.docs]
+command = "cargo"
+args = ["doc", "--open"]
+dependencies = ["build"]
+
+[tasks.uml]
+command = "open"
+args = ["-a", "Google Chrome", "./diagrams/ml.svg"]
```
Comment (jamesray1) [inline Makefile.toml:104]: Why have Google Chrome here? I use Firefox.

**Comment by jamesray1** [comment]
Looks good, I'll merge.

File: Makefile.toml (hunk 1 above)
Comment (ltfschoen) [reply Makefile.toml:104]: Good point. I'll add it as an issue

Comment (jamesray1) [reply Makefile.toml:104]: Thanks.

**Pull request merged**